### PR TITLE
a bit more robust

### PR DESCRIPTION
**Pull request by raymondng**
Add try-catch; fallback to secure social
@andrewconner @eishay @2is10

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/controller/UserActions.scala [hunk 1]
```diff
@@ -112,14 +112,19 @@ trait SecureSocialHelper extends Logging {
...
-trait UserActionsHelper extends UserActionsRequirements {
+trait UserActionsHelper extends UserActionsRequirements with Logging {
 
   protected def getUserIdFromRequest(implicit request: Request[_]): Option[Id[User]] = {
     request.session.get(KifiSession.FORTYTWO_USER_ID).map(id => Id[User](id.toLong))
   }
 
   def getUserIdOpt(implicit request: Request[_]): Future[Option[Id[User]]] = {
-    getUserIdFromRequest match {
+    val kifiIdOpt = try { getUserIdFromRequest } catch {
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/controller/UserActions.scala:122]: I assume this is to protect against `FORTYTWO_USER_ID` not being a long? What about just returning `None` from `getUserIdFromRequest`?

File: kifi-backend/modules/common/app/com/keepit/common/controller/UserActions.scala [hunk 2]
```diff
@@ -112,14 +112,19 @@ trait SecureSocialHelper extends Logging {
...
+      case t: Throwable =>
+        log.error(s"[getUserIdFromRequest] Caught exception $t while retrieving userId from request; cause=${t.getCause}", t)
+        None
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/controller/UserActions.scala:125]: This should Airbrake. It's quite serious and we don't want to not know it's happening.

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/controller/UserActions.scala:125]: Agreed. Airbrake is not yet available in these helpers -- it's on the TODO list.

Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/controller/UserActions.scala:125]: Hm, that's unfortunate.

File: kifi-backend/modules/common/app/com/keepit/common/controller/UserActions.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/controller/UserActions.scala:122]: Yes that's also a possibility that I considered, but that'd hide it from the caller. Airbrake would have been indeed handy but it's not available (yet). To make it available, I'll need to rework things a bit.

**Pull request merged**